src/assets/CollectiblesController.ts

```diff
@@ -1148,6 +1153,105 @@ export class CollectiblesController extends BaseController<
 
     this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
   }
+
+  /**
+   * Update colletible isTransacted status
+   *
+   * @param address - Hex address of the collectible contract.
+   * @param tokenId - Number that represents the id of the token.
+   * @param isTransacting - Collectible transacting status.
+   * @param transactionId - Hex id of the send transaction of the collectible.
+   */
+  updateCollectibleTransactionStatus(
+    address: string,
+    tokenId: string,
+    isTransacting: boolean,
+    transactionId: string,
+  ) {
+    const { allCollectibles } = this.state;
+    const { chainId, selectedAddress } = this.config;
+    const collectibles = allCollectibles[selectedAddress]?.[chainId] || [];
+    const index: number = collectibles.findIndex(
+      (collectible) =>
+        collectible.address === address && collectible.tokenId === tokenId,
+    );
+
+    if (index === -1) {
+      return;
+    }
+    const updatedCollectible: Collectible = {
+      ...collectibles[index],
+      isTransacting,
+      transactionId,
+    };
+
+    // Update Collectibles array
+    collectibles[index] = updatedCollectible;
+
+    this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
+  }
+  /**
+   * Reset the collectible transaction id value and turn false the isTransacting prop
+   *
+   * @param transactionId - Hex id of the send transaction of the collectible.
+   * @returns
+   */
+  resetCollectibleTransactionStatusByTransactionId(
+    transactionId: string,
+  ): boolean {
+    const { allCollectibles } = this.state;
+    const { chainId, selectedAddress } = this.config;
+    const collectibles = allCollectibles[selectedAddress]?.[chainId] || [];
+    const index: number = collectibles.findIndex(
+      (collectible) => collectible.transactionId === transactionId,
+    );
+
+    if (index === -1) {
+      return false;
+    }
+    const updatedCollectible: Collectible = {
+      ...collectibles[index],
+      isTransacting: false,
+      transactionId: '',
+    };
+
+    // Update Collectibles array
+    collectibles[index] = updatedCollectible;
+
+    this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
+    return true;
+  }
+
+  /**
+   * Returns if the token is being transacted
+   *
+   * @param address - Hex address of the collectible contract.
+   * @param tokenId - Number that represents the id of the token.
+   * @returns
+   */
+  getCollectibleTransactionStatus(
+    address: string,
+    tokenId: string,
+  ):
+    | { isTransacting: boolean | undefined; transactionId: string | undefined }
```

Comment:
```suggestion
    | { isTransacting: boolean; transactionId: string | undefined }
```

You can update the logic to always return a boolean instead of `undefined`.